src/libAtomVM/nifs.c

```diff
@@ -4088,11 +4088,25 @@ static term nif_erlang_group_leader(Context *ctx, int argc, term argv[])
             RAISE_ERROR(BADARG_ATOM);
         }
 
-        if (term_is_local_pid(leader)) {
-            // We cannot put leader term on the heap
+        if (target != ctx) {
+            // always use signals when changing group leader of another context
```

Comment:
We should only lock the process table in this case. If it's ctx, there is no need to lock anything.

So I suggest testing ctx->process_id with local_process_id before calling globalcontext_get_process_lock.

Reply:
Good idea :)